core/services/gateway/handlers/capabilities/handler.go

```diff
@@ -182,12 +216,137 @@ func (h *handler) handleWebAPIOutgoingMessage(ctx context.Context, msg *api.Mess
 	return nil
 }
 
+//	body := api.MessageBody{
+//		MessageId: types.RandomID().String(),
+//		DonId:     h.connector.DonID(),
+//		Method:    webapicapabilities.MethodWebAPITriggerUpdateMetadata,
+//		Receiver:  gatewayIDStr,
+//		Payload:   payloadJSON,
+//	}
+func (h *handler) handleWebAPITriggerUpdateMetadata(ctx context.Context, msg *api.Message, nodeAddr string) error {
+	body := msg.Body
+	h.lggr.Debugw("handleWebAPITriggerUpdateMetadata", "body", body, "payload", string(body.Payload))
+
+	var payload map[string]string
+	err := json.Unmarshal(body.Payload, &payload)
+	donID := body.DonId
+	if err != nil {
+		h.lggr.Errorw("error decoding payload", "err", err, "payload", string(body.Payload))
+		return err
+	}
+	triggersConfig := make(map[string]webapicap.TriggerConfig)
+	for triggerID, configPbString := range payload {
+		pbBytes, err := base64.StdEncoding.DecodeString(configPbString)
+		if err != nil {
+			h.lggr.Errorw("error decoding pb bytes", "err", err)
+			return err
+		}
+		pb := &pb.Map{}
+		err = proto.Unmarshal(pbBytes, pb)
+		if err != nil {
+			h.lggr.Errorw("error unmarshalling", "err", err)
+			return err
+		}
+		vmap, err := values.FromMapValueProto(pb)
+		if err != nil {
+			h.lggr.Errorw("error FromMapValueProto", "err", err)
+			return err
+		}
+		if vmap == nil {
+			h.lggr.Errorw("error FromMapValueProto nil vmap")
+			return err
+		}
+		reqConfig, err := h.ValidateConfig(vmap)
+		if err != nil {
+			h.lggr.Errorw("error validating config", "err", err)
+			return err
+		}
+		triggersConfig[triggerID] = *reqConfig
+	}
+	h.triggersConfig.triggersConfigMap[donID] = TriggersConfig{lastUpdatedAt: time.Now(), triggersConfig: triggersConfig}
+	return nil
+}
+
+type triggerStruct struct {
+	config webapicap.TriggerConfig
+	count  uint
+}
+
+func (h *handler) updateTriggerConsensus() {
```

Comment:
looks like this function is not called in this PR. how do we determine when we should trigger consensus?

Reply:
Good question. I forgot to put a TODO that once i have a timer with tests working (from the trigger), the handler needs to do something similar.  I've updated the PR description.